mount.go

```diff
@@ -96,6 +76,68 @@ func Mount(
 		err = fmt.Errorf("mount (background): %v", err)
 		return
 	}
+	return
+}
+
+// FuseMount attempts to mount a file system on the given directory. It can block
+// fuse server setup completion using supplied ready channel.
+func FuseMount(
+	dir string,
+	config *MountConfig,
+	ready chan error) (dev uintptr, err error) {
+	// Sanity check: make sure the mount point exists and is a directory. This
+	// saves us from some confusing errors later on OS X.
+	fi, err := os.Stat(dir)
+	switch {
+	case os.IsNotExist(err):
+		return
+
+	case err != nil:
+		err = fmt.Errorf("Statting mount point: %v", err)
+		return
+
+	case !fi.IsDir():
+		err = fmt.Errorf("Mount point %s is not a directory", dir)
+		return
+	}
+
+	// Begin the mounting process, which will continue in the background.
+	dev, err = mount(dir, config, ready)
+	if err != nil {
+		err = fmt.Errorf("mount: %v", err)
+		return
+	}
+	return
+}
+
+func IsMounted(
```

Comment:
Why do you need this? Please also add documentation.

Reply:
This is just a util func and some still do not need this. Now I'm wondering this should be implemented in client side, not in this jacobsa/fuse.git. Perhaps I'll remove these.